src/generator/AutoRest.Ruby/Model/MethodRb.cs

```diff
@@ -606,6 +606,16 @@ public string GetDeserializationString(IModelType type, string valueReference =
                 builder.AppendLine("{1} = @client.deserialize(result_mapper, {0}, '{1}')", responseVariable, valueReference);
             }
 
+            if (type is CompositeType)
+            {
+                var composite = type as CompositeType;
+                if (composite.BaseModelType != null && composite.BaseModelType.Name == "Resource")
```

Comment:
Agreed, though with this question I was wondering if we could avoid writing these lines of code in the file if there's no resource_group_name coming as parameter of the method. For example the following operation would end up with this added line of code, but it really doesn't apply to the method. Operation: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-compute/2016-03-30/swagger/compute.json#L294, which in generated code looks like this https://github.com/Azure/azure-sdk-for-ruby/blob/master/management/azure_mgmt_compute/lib/generated/azure_mgmt_compute/virtual_machine_extension_images.rb#L163 
I understand it would still work because the property is not defined, but it's really not applicable.